### PR TITLE
329 add storage to get/delete device endpoints

### LINKED ISSUE #15: Storage Provider
#2

Create a storage provider that can use either a local provider or an azure provider based on a configuration file. They will use a shared interface so that the actual process of performing crud operations are abstracted out. In addition the storage provider should make use of dependency injection to help facilitate this.

- [x] As a developer, based on a configuration file, I should be able to get or put a file in azure storage.
- [x] As a developer, based on a configuration file, I should be able to get or put a file on my system.
- [x] As a developer, I should have a service that I can use to manage interacting with storage.
- [x] Research how to upload a file and what endpoints to attach them to.
- [ ] Hook storage provider into existing endpoints

### LINKED ISSUE #329: Update Get/Delete device endpoints to use storage provider
References user story #15
References #228

Add storage provider to the get/device endpoints. On a get call, the returned files and photos should have their fileUrl set to a presigned url. On a delete call, files associated with a device should be removed from storage.

### PR DESCRIPTION
**Pull request by Kenny-Miller**
Closes #329
Closes #15
References #228 

Added the storage provider to the get and delete endpoints.

### CODE REVIEW

File: BEIMA.Backend/DeviceFunctions/DeleteDevice.cs [hunk 1]
```diff
@@ -36,11 +41,35 @@ public static class DeleteDevice
             var mongo = MongoDefinition.MongoInstance;
             var deviceId = new ObjectId(id);
```
Comment (josephmoore99) [inline BEIMA.Backend/DeviceFunctions/DeleteDevice.cs:42]: Looks like I must've missed this, but we should run an ObjectId.TryParse on the id before assigning it here, and return the invalid id message if it's not good.

**Comment by josephmoore99** [comment]
Excited to finally have all the files stuff implemented!

File: BEIMA.Backend/DeviceFunctions/DeleteDevice.cs (hunk 1 above)
Comment (Kenny-Miller) [reply BEIMA.Backend/DeviceFunctions/DeleteDevice.cs:42]: It does get try parsed right above line 36. I can change the assignment to ObjetId.Parse(id) though if that's better

Comment (josephmoore99) [reply BEIMA.Backend/DeviceFunctions/DeleteDevice.cs:42]: Oh, woops, didn't see that, should be fine then.

**Review by josephmoore99 — APPROVED** [review]

**Review by AshlynAdamson — APPROVED** [review]

**Review by roguedeleter — APPROVED** [review]

**Pull request merged**